src/learn/internal/learnservice.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "learnservice.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QBuffer>
+
+#include "log.h"
+
+using namespace mu::learn;
+using namespace mu::network;
+using namespace mu::async;
+
+LearnService::LearnService(QObject* parent)
+    : QObject(parent)
+{
+}
+
+Promise<Playlist> LearnService::startedPlaylist() const
+{
+    return Promise<Playlist>([this](Promise<Playlist>::Resolve resolve,
+                                    Promise<Playlist>::Reject) {
+        QUrl playlistUrl = configuration()->startedPlaylistUrl();
+        if (playlistUrl.isEmpty()) {
+            return resolve({});
+        }
+
+        resolve(requestPlaylist(playlistUrl));
+    });
+}
+
+Promise<Playlist> LearnService::advancedPlaylist() const
+{
+    return Promise<Playlist>([this](Promise<Playlist>::Resolve resolve,
+                                    Promise<Playlist>::Reject) {
+        QUrl playlistUrl = configuration()->advancedPlaylistUrl();
+        if (playlistUrl.isEmpty()) {
+            return resolve({});
+        }
+
+        resolve(requestPlaylist(playlistUrl));
+    });
+}
+
+void LearnService::openVideo(const std::string& videoId) const
+{
+    QUrl videoUrl = configuration()->videoOpenUrl(videoId);
+    interactive()->openUrl(videoUrl.toString().toStdString());
+}
+
+Playlist LearnService::requestPlaylist(const QUrl& playlistUrl) const
+{
+    network::INetworkManagerPtr networkManager = networkManagerCreator()->makeNetworkManager();
+    RequestHeaders headers = configuration()->headers();
+
+    QBuffer playlistItemsData;
+    Ret playlistItems = networkManager->get(playlistUrl, &playlistItemsData, headers);
+    if (!playlistItems) {
+        LOGE() << playlistItems.toString();
+        return {};
+    }
+
+    QVariantMap playlistInfo = QJsonDocument::fromJson(playlistItemsData.data()).toVariant().toMap();
+
+    std::vector<std::string> playlistItemsIds = parsePlaylistItemsIds(playlistInfo);
```

Comment:
I believe it's better to parse JSON directly, without converting to a QVariant(Map)